apps/hash-ai-worker-ts/src/activities/shared/get-llm-response/llm-message.ts

```diff
@@ -0,0 +1,275 @@
+import type {
+  ChatCompletionAssistantMessageParam,
+  ChatCompletionMessageParam as OpenAiMessage,
+  ChatCompletionToolMessageParam,
+  ChatCompletionUserMessageParam,
+} from "openai/resources";
+
+import type { AnthropicMessage } from "./anthropic-client";
+
+export type LlmMessageTextContent = {
+  type: "text";
+  text: string;
+};
+
+export type LlmMessageToolUseContent<ToolName = string> = {
+  type: "tool_use";
+  id: string;
+  name: ToolName;
+  input: object;
+};
+
+export type LlmAssistantMessage<ToolName = string> = {
+  role: "assistant";
+  content: (LlmMessageTextContent | LlmMessageToolUseContent<ToolName>)[];
+};
+
+export type LlmMessageToolResultContent = {
+  type: "tool_result";
+  tool_use_id: string;
+  content?: string;
+  is_error?: true;
+};
+
+export type LlmUserMessage = {
+  role: "user";
+  content: (LlmMessageTextContent | LlmMessageToolResultContent)[];
+};
+
+export type LlmMessage = LlmAssistantMessage | LlmUserMessage;
+
+type LlmMessageContent = (
+  | LlmMessageTextContent
+  | LlmMessageToolUseContent
+  | LlmMessageToolResultContent
+)[];
+
+export const mapLlmMessageToAnthropicMessage = (params: {
+  message: LlmMessage;
+}): AnthropicMessage => ({
+  role: params.message.role,
+  content: params.message.content,
+});
```

Comment:
In its current form, yes because the current `LlmMessage` format matches the one from Anthropic. If we deviate from it this method may be required though, so have added it for consistency for now.